File: Backend/Microservices/Expense/EMS.Core/Entities/Category.cs
using EMS.Core.Entities.Common;
using EMS.Core.Enums;

namespace EMS.Core.Entities
{
    public class Category : BaseAuditableEntity<int>
    {
        public string? UserId { get; set; }
        public string Name { get; set; } = default!;
        public CategoryType Type { get; set; }
        public TransactionType FinancialFlowType { get; set; } // Whether a category is for expense or income
        public Guid? IconId { get; set; }

        // Navigations
        public ICollection<Transaction> Transactions { get; set; } = [];
        public Media? Icon { get; set; }
        public ICollection<ExtractedTransaction> ExtractedTransactions { get; set; } = [];
    }
}

File: Backend/Microservices/AI/app/schemas/category.py
from enum import Enum
from uuid import UUID
from pydantic import BaseModel

from app.schemas.transaction import TransactionType


class CategoryType(Enum):
    Default = "Default"
    Custom = "Custom"


class Category(BaseModel):
    id: int
    name: str
    is_default: bool
    type: CategoryType
    financial_flow_type: TransactionType
    icon_id: UUID | None = None
    icon_url: str | None = None
